engine/libstats.go

```diff
@@ -234,15 +234,48 @@ func (sq *StatQueue) TenantID() string {
 
 // ProcessEvent processes a utils.CGREvent, returns true if processed
 func (sq *StatQueue) ProcessEvent(tnt, evID string, filterS *FilterS, evNm utils.MapStorage) (err error) {
+	var oneEv bool
+	if oneEv, err = sq.isOneEvent(tnt, filterS, evNm); oneEv {
+		return err
+	}
 	if _, err = sq.remExpired(); err != nil {
 		return
 	}
+
 	if err = sq.remOnQueueLength(); err != nil {
 		return
 	}
 	return sq.addStatEvent(tnt, evID, filterS, evNm)
 }
 
+func (sq *StatQueue) isOneEvent(tnt string, filterS *FilterS, evNm utils.MapStorage) (bool, error) {
```

Comment:
isOneEvent should only confirm if we are using special processing or not. There should be no other logic within.